app/javascript/packages/analytics/index.ts

```diff
@@ -17,14 +17,30 @@ interface NewRelicGlobals {
  */
 export async function trackEvent(event: string, payload?: object): Promise<void> {
   const endpoint = getConfigValue('analyticsEndpoint');
+
   if (!endpoint) {
     return;
   }
 
+  const eventJson = JSON.stringify({ event, payload });
+
+  // Favor making analytics requests using sendBeacon(), which can be prioritized
+  // appropriately by the browser and have a better chance of succeeding during page unload.
+  try {
```

Comment:
I had never heard of `sendBeacon` before, this is really cool! And it solves one of the bigger issues with this function having to be `async` to allow the analytics request to be complete.

Based on browser support, I personally feel it'd make a lot of sense to avoid the fallback altogether to be able to simplify this method to be synchronous. There's a handful of places where we have to jump through hoops to wait for the asynchronous request to finish that we'd be able to simplify by doing so.

Reply:
Some code which would benefit:

https://github.com/18F/identity-idp/blob/b121550fa7a3d9096ddbed34a8c13f77fe3649b0/app/javascript/packages/document-capture/components/barcode-attention-warning.tsx#L31

https://github.com/18F/identity-idp/blob/b121550fa7a3d9096ddbed34a8c13f77fe3649b0/app/javascript/packages/document-capture-polling/index.ts#L86

These also benefit, and are likely currently broken honestly because they don't (but should) await the result:

https://github.com/18F/identity-idp/blob/b121550fa7a3d9096ddbed34a8c13f77fe3649b0/app/javascript/packages/document-capture/components/in-person-location-post-office-search-step.tsx#L78

https://github.com/18F/identity-idp/blob/b121550fa7a3d9096ddbed34a8c13f77fe3649b0/app/javascript/packages/countdown/countdown-alert-element.ts#L36

Reply:
`:chef-kissing-fingers:`

```ts
export function trackEvent(event: string, payload?: object): void {
  const endpoint = getConfigValue('analyticsEndpoint');
  if (endpoint) {
    const blob = new Blob([JSON.stringify({ event, payload })], { type: 'application/json' });
    navigator.sendBeacon(endpoint, blob);
  }
}
```

Reply:
Yeah I'm on board with removing the fetch fallback -- see [ff30cc7](https://github.com/18F/identity-idp/pull/8496/commits/ff30cc7b5bbdc1406c6c4ede5e85879eab72ca5b).